### ISSUE TITLE
NVidia Vsync ExceptionInInitializerError

### ISSUE THREAD

**Issue by CoWayger** [libgdx/libgdx#2743]
Hello, I'm facing a problem, yet solution is very simple...

I have installed nvidia optimus graphics on my system, yet vsync is not supported by nvidia driver, libgdx throwns:

Exception in thread "LWJGL Application" java.lang.ExceptionInInitializerError
	at com.badlogic.gdx.backends.lwjgl.LwjglGraphics.setVSync(LwjglGraphics.java:434)
	at com.badlogic.gdx.backends.lwjgl.LwjglApplication$1.run(LwjglApplication.java:112)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 0
	at org.lwjgl.opengl.LinuxDisplay.getAvailableDisplayModes(LinuxDisplay.java:954)
	at org.lwjgl.opengl.LinuxDisplay.init(LinuxDisplay.java:738)
	at org.lwjgl.opengl.Display.<clinit>(Display.java:138)
	... 2 more
AL lib: (EE) alc_cleanup: 1 device not closed

The problem is on this line in LinuxDisplay.java (line 954)

Screen[] resolutions = XRandR.getResolutions(XRandR.getScreenNames()[0]);

You are using 0th element from returned array, unfortunately, on vsync not working platforms it returns null. 

You should check size of returned array.

**Comment by xoppa** [comment]
I'm afraid there's not much we can do about that. Please consider reporting an issue for lwjgl (which included the LinuxDisplay.java file).

**Comment by ImDevinC** [comment]
Just a quick note here, I was having this same problem and it was actually just because I didn't have xrandr installed. As soon as I installed xrandr on my system, my program ran properly.

**Comment by johannesmueller-fr** [comment]
Since I faced the same problem today, I checked where the real bug hides:
in my case this behaviour is triggered because of my dual-gpu-setup. Since PRIME got introduced my system suspects two screens (LVDS and LVDS-1-1)
Since LVDS-1-1 is not a real screen, but the same one as LVDS xrandr -q returns simply "LVDS-1-1 connected".

This string is not expected by the org.lwjgl.opengl.XRandR.populate() method.
So I inserted ths snippet in line 95:

          if ( sa.length <= 2 ) {
            parseScreenHeader(currentScreenPosition, "");
            continue;
          }

Therefore sa[2] won't be called... and it works... but it's a pain in the ass to compile all the libraries again :D
